### PR TITLE
9588 to test

### PR DESCRIPTION
**Pull request by TomElliottFlexion**

### CODE REVIEW

File: shared/src/business/useCases/practitioners/deletePractitionerDocumentInteractor.test.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    });
+    expect(
+      applicationContext.getPersistenceGateway().deleteDocumentFromS3.mock
+        .calls[0][0],
+    ).toMatchObject({
+      key: practitionerDocumentFileId,
+    });
+  });
+
+  it('should create a new practitioner document', async () => {
```
Comment (rgoshen) [inline shared/src/business/useCases/practitioners/deletePractitionerDocumentInteractor.test.ts]: I am confused and sorry for my ignorance. If this interactor is meant to delete the document, then why would it turn around and create it?

File: web-api/src/practitioners/editPractitionerDocumentLambda.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const { genericHandler } = require('../genericHandler');
+
+/**
+ * creates a practitioner document for a practitioner
```
Comment (rgoshen) [inline web-api/src/practitioners/editPractitionerDocumentLambda.js]: Nitpik- does this create or edit?

File: shared/src/business/useCases/practitioners/editPractitionerDocumentInteractor.test.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  it('should throw a validation error if the practitioner document has the wrong type', async () => {
+    await expect(
+      editPractitionerDocumentInteractor(applicationContext, {
+        barNumber,
+        documentMetadata: { ...mockDocumentMetadata, categoryType: 'GG' },
+      }),
+    ).rejects.toThrow(InvalidEntityError);
+  });
+
+  it('should throw a validation error if the practitioner document is missing information', async () => {
```
Comment (rgoshen) [inline shared/src/business/useCases/practitioners/editPractitionerDocumentInteractor.test.ts:49]: How would file name be undefined from an edit standpoint?  You changed the stateDrivenFileInput to default to existing filename.

File: web-client/src/views/FileDocument/StateDrivenFileInput.jsx [hunk 4]
```diff
@@ -75,7 +75,9 @@ export const StateDrivenFileInput = connect(
             <span className="success-message icon-upload margin-right-1">
               <FontAwesomeIcon icon="check-circle" size="1x" />
             </span>
-            <span className="mr-1">{fileOnForm.name}</span>
+            <span className="mr-1">
+              {fileOnForm.name || form.existingFileName}
```
Comment (rgoshen) [inline web-client/src/views/FileDocument/StateDrivenFileInput.jsx:79]: I like this change!

File: shared/src/business/useCases/practitioners/deletePractitionerDocumentInteractor.test.ts (hunk 1 above)
Comment (TomElliottFlexion) [reply shared/src/business/useCases/practitioners/deletePractitionerDocumentInteractor.test.ts]: Didn't update the test name, my bad!

File: web-api/src/practitioners/editPractitionerDocumentLambda.js (hunk 2 above)
Comment (TomElliottFlexion) [reply web-api/src/practitioners/editPractitionerDocumentLambda.js]: Good catch

File: shared/src/business/useCases/practitioners/editPractitionerDocumentInteractor.test.ts (hunk 3 above)
Comment (TomElliottFlexion) [reply shared/src/business/useCases/practitioners/editPractitionerDocumentInteractor.test.ts:49]: In this case, `existingFileName` is only for display purposes, and the `fileName` on the edited document is already set, unless it is replaced during editing

**Review by rgoshen — APPROVED** [review]

**Pull request merged**